great_expectations/data_context/data_context.py

```diff
@@ -192,22 +225,20 @@ def get_available_data_asset_names(self, datasource_names=None, generator_names=
 
         for idx, datasource_name in enumerate(datasource_names):
             datasource = self.get_datasource(datasource_name)
-            data_asset_names.append(
-                {
-                    "datasource": datasource_name,
-                    "generators": datasource.get_available_data_asset_names(generator_names[idx] if generator_names is not None else None)
-                }
-            )
+            data_asset_names[datasource_name] = \
+                datasource.get_available_data_asset_names(generator_names[idx] if generator_names is not None else None)
         return data_asset_names
 
-    def get_batch(self, datasource_name, data_asset_name, batch_kwargs=None, **kwargs):
-        data_asset_name = self._normalize_data_asset_name(data_asset_name)
-        # datasource_name = find(data_asset_name.split("/")[0]
-        datasource = self.get_datasource(datasource_name)
+    def get_batch(self, data_asset_name, batch_kwargs=None, **kwargs):
```

Comment:
The expectation suite is available among the ways of creating a normalized data_asset_name:

data_asset_name ==
datasource / generator / generator_asset / expectation_suite

I have updated the logic to allow both:
(a) datasource / generator_asset, and
(b) generator_asset / expectation_suite
combinations.

NOTE: while this will make it easier to create and access expectation suites in cases where there are few datasources, it will in general make using the two-part name scheme harder since there are more ambiguities.

NOTE2: the guessing logic is expensive (it creates lists of all available assets and iterates over them), maybe to the point of impossible for generators that have lots of assets or are otherwise slow (think generator introspecting database)

That seems like a worthwhile tradeoff at this point since it is easy to get started and the default generators are tame anyway.